src/api/cvc4cppkind.h

```diff
@@ -1832,11 +1832,23 @@ enum CVC4_PUBLIC Kind : int32_t
    */
   MEMBER,
   /**
-   * The set of the single element given as a parameter.
+   * From the API, use Term Solver::mkSingleton(Sort s, Term t)
```

Comment:
With `mkSingleton` we don't need to expose SINGLETON in the api. It should only be internal. However, it is used by the parser. So we can add a special token for it. 
Alternatively we can expose it  in the API and remove mkSingleton as discussed below.

Reply:
We need SINGLETON in the API since getting the kind of a singleton node should return SINGLETON.